### DISCUSSION TITLE
Install PyMuPDF

### DISCUSSION THREAD

**Discussion by dayanych** [pymupdf/PyMuPDF#2394] [Looking for help]
I was installing requirements, where the error popped up 
>ERROR: Failed to build wheels for PyMuPDF, which is required to install projects based on pyproject.toml

I tried to install the PyMuPDF with python setup.py install, but then: 

>Traceback (most recent call last):
> File "setup.py", line 706, in <module>
>    subprocess.run( command, shell=True, check=True)
> File "C:\Users\we\AppData\Local\Programs\Python\Python36\lib\subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
>subprocess.CalledProcessError: Command 'cd mupdf-1.22.0-source&&"devenv.com" >platform/win32/mupdf.sln /Build "ReleaseTesseract|x64"/Project mupdf' returned non-zero exit status 1.

I used "python -m pip install --upgrade pip" and "python -m pip install --upgrade pymupdf". but after "python -m pip install --upgrade pymupdf" i got:

>Running setup.py clean for pymupdf
>Failed to build pymupdf
>Installing collected packages: pymupdf
>    Running setup.py install for pymupdf ... error

What should I do? pip 23.1.2 from C:\Users\we\AppData\Local\Programs\Python\Python311\Lib\site-packages\pip (python 3.11)

**Comment by julian-smith-artifex-com** [comment]
It looks like you're running Windows but for some reason the Wheels on pypi.org do not match your OS or CPU. So pip is trying to build PyMuPDF from source, but PyMuPDF/setup.py cannot find devenv.com (which is part of Microsoft Visual Studio).

So we'll need a little more information here.

What OS are you using?
What CPU do you have?
Do you have Microsoft Visual Studio installed; if yes, what version?

Finally, could you paste the entire output from your pip command?
